homeassistant/components/medcom_ble/sensor.py

```diff
@@ -0,0 +1,106 @@
+"""Support for Medcom BLE radiation monitor sensors."""
+from __future__ import annotations
+
+import logging
+
+from medcom_ble import MedcomBleDevice
+
+from homeassistant import config_entries
+from homeassistant.components.sensor import (
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import CONNECTION_BLUETOOTH, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import DOMAIN, UNIT_CPM
+
+_LOGGER = logging.getLogger(__name__)
+
+SENSORS_MAPPING_TEMPLATE: dict[str, SensorEntityDescription] = {
+    "cpm": SensorEntityDescription(
+        key="cpm",
+        translation_key="cpm",
+        native_unit_of_measurement=UNIT_CPM,
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:radioactive",
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: config_entries.ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Medcom BLE radiation monitor sensors."""
+
+    coordinator: DataUpdateCoordinator[MedcomBleDevice] = hass.data[DOMAIN][
+        entry.entry_id
+    ]
+
+    entities = []
+    _LOGGER.debug("got sensors: %s", coordinator.data.sensors)
+    for sensor_type, sensor_value in coordinator.data.sensors.items():
+        if sensor_type not in SENSORS_MAPPING_TEMPLATE:
+            _LOGGER.debug(
+                "Unknown sensor type detected: %s, %s",
+                sensor_type,
+                sensor_value,
+            )
+            continue
+        entities.append(
+            MedcomSensor(
+                coordinator, coordinator.data, SENSORS_MAPPING_TEMPLATE[sensor_type]
```

Comment:
It would be cleaner to not pass `coordinator.data` here since you can already extract the device in `__init__`

Reply:
good point yes